plugin/trino-base-jdbc/src/main/java/io/trino/plugin/jdbc/ColumnWithAliasFormatterFixed.java

```diff
@@ -0,0 +1,46 @@
+/*
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.trino.plugin.jdbc;
+
+import io.trino.spi.connector.ConnectorSession;
+
+import java.util.Optional;
+
+import static com.google.common.base.Splitter.fixedLength;
+import static io.trino.plugin.jdbc.JdbcMetadataSessionProperties.COLUMN_ALIAS_MAX_CHARS;
+
+public class ColumnWithAliasFormatterFixed
+        implements ColumnWithAliasFormatter
+{
+    public static final int DEFAULT_COLUMN_ALIAS_LENGTH = 30;
+    public static final int ORIGINAL_COLUMN_NAME_LENGTH = 24;
+
+    @Override
+    public JdbcColumnHandle format(JdbcColumnHandle column, int nextSyntheticColumnId, ConnectorSession session)
+    {
+        Integer property = Optional.ofNullable(session.getProperty(COLUMN_ALIAS_MAX_CHARS, Integer.class))
+                .orElse(DEFAULT_COLUMN_ALIAS_LENGTH);
+        int sequentialNumberLength = property - ORIGINAL_COLUMN_NAME_LENGTH - 1;
+
+        String originalColumnNameTruncated = fixedLength(ORIGINAL_COLUMN_NAME_LENGTH)
+                .split(column.getColumnName())
+                .iterator()
+                .next();
+        String formatString = "%s_%0" + sequentialNumberLength + "d";
```

Comment:
instead of a format you can just concatenate string with padded nextSyntheticColumnId as follows:

`String columnName = originalColumnNameTruncated + "_" + padStart(Integer.toString(nextSyntheticColumnId), sequentialNumberLength, '0'));`

padStart comes from Guava's Strings class

Reply:
Like any programming, you can do this in 1000 different ways. Can you explain better what is the advantage of using the approach you proposed? Is there a disadvantage of using what I proposed?

Reply:
`String.format` generates a lot of garbage and is less performant so we avoid it on the hot paths (see: https://github.com/trinodb/trino/pull/15258 https://github.com/trinodb/trino/pull/15316)

Reply:
The solution that I've proposed is free of these problems and inlines nicely.

Reply:
because existing code is imperfect doesn't mean future code should be do.

The good thing to do here would be to send follow-up PR which fixes the existing String.format usages.

Reply:
IMO "premature optimization is the root of all evil" doesn't mean that we shouldn't to the obvious things that we already know that we can do better.

Replying to your `ownership` comment this class was created as a copy during: https://github.com/trinodb/trino/pull/10059. I'm not the sole author of this code but I can always improve it: https://github.com/trinodb/trino/pull/18885

Reply:
Changed String.format() to guava's pad method.